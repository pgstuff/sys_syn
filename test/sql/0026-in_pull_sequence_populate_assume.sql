BEGIN;

CREATE EXTENSION tinyint
    SCHEMA public;

CREATE EXTENSION sys_syn;

CREATE SCHEMA user_data
    AUTHORIZATION postgres;

CREATE TABLE user_data.test_table (
        test_table_key integer NOT NULL,
        test_table_text text,
        CONSTRAINT test_table_pkey PRIMARY KEY (test_table_key));

INSERT INTO sys_syn.in_groups_def VALUES ('in');

DO $$BEGIN
    EXECUTE sys_syn.in_table_add_sql('user_data.test_table'::regclass, 'in');
END$$;

SELECT sys_syn.in_pull_sequence_populate_assume();

SELECT * FROM sys_syn.in_pull_sequences_def;

SELECT * FROM sys_syn.in_pull_sequence_pulls;

ROLLBACK;
