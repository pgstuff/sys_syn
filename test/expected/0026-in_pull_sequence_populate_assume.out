BEGIN;
CREATE EXTENSION tinyint SCHEMA public;
CREATE EXTENSION sys_syn;
CREATE SCHEMA user_data
    AUTHORIZATION postgres;
CREATE TABLE user_data.test_table (
        test_table_id integer NOT NULL,
        test_table_text text,
        CONSTRAINT test_table_pid PRIMARY KEY (test_table_id));
INSERT INTO sys_syn.in_groups_def VALUES ('in');
DO $$BEGIN
    EXECUTE sys_syn.in_table_create_sql('user_data.test_table'::regclass, 'in');
END$$;
SELECT sys_syn.in_pull_sequence_populate_assume();
 in_pull_sequence_populate_assume 
----------------------------------
 
(1 row)

SELECT * FROM sys_syn.in_pull_sequences_def;
 in_pull_sequence_id | comments 
---------------------+----------
 in                  | 
(1 row)

SELECT * FROM sys_syn.in_pull_sequence_pulls;
 in_pull_sequence_id | sequence_index | in_pull_id | comments 
---------------------+----------------+------------+----------
 in                  |              1 | test_table | 
(1 row)

ROLLBACK;
